var/spack/repos/builtin/packages/cabana/package.py

```diff
@@ -4,47 +4,58 @@
 
 
 class Cabana(CMakePackage):
     """The Exascale Co-Design Center for Particle Applications Toolkit
     """
     homepage = "https://github.com/ECP-copa/Cabana"
     git      = "https://github.com/ECP-copa/Cabana.git"
-    url      = "https://github.com/ECP-copa/Cabana/archive/0.1.0.tar.gz"
+    url      = "https://github.com/ECP-copa/Cabana/archive/0.4.0.tar.gz"
+
+    maintainers = ["junghans", "sslattery", "streeve"]
 
     version('master', branch='master')
+    version('0.4.0', sha256='c347d23dc4a5204f9cc5906ccf3454f0b0b1612351bbe0d1c58b14cddde81e85')
     version('0.3.0', sha256='fb67ab9aaf254b103ae0eb5cc913ddae3bf3cd0cf6010e9686e577a2981ca84f')
     version('0.2.0', sha256='3e0c0e224e90f4997f6c7e2b92f00ffa18f8bcff72f789e0908cea0828afc2cb')
     version('0.1.0', sha256='3280712facf6932b9d1aff375b24c932abb9f60a8addb0c0a1950afd0cb9b9cf')
     version('0.1.0-rc0', sha256='73754d38aaa0c2a1e012be6959787108fec142294774c23f70292f59c1bdc6c5')
 
-    variant('serial', default=True, description="enable Serial backend (default)")
-    variant('openmp', default=False, description="enable OpenMP backend")
-    variant('cuda', default=False, description="enable Cuda backend")
+    kokkos_backends = Kokkos.devices_variants
+    for backend in kokkos_backends:
+        deflt, descr = kokkos_backends[backend]
+        variant(backend.lower(), default=deflt, description=descr)
+
     variant('shared', default=True, description='Build shared libraries')
     variant('mpi', default=True, description='Build with mpi support')
+    variant('arborx', default=False, description='Build with ArborX support')
+    variant('heffte', default=False, description='Build with heFFTe support')
+    variant('hypre', default=False, description='Build with HYPRE support')
 
     depends_on("cmake@3.9:", type='build')
-    depends_on("kokkos-legacy+serial", when="@:0.2.0+serial")
-    depends_on("kokkos-legacy+openmp", when="@:0.2.0+openmp")
-    depends_on("kokkos-legacy+cuda", when="@:0.2.0+cuda")
-    depends_on("kokkos@3.1:+serial", when="@0.3.0:+serial")
-    depends_on("kokkos@3.1:+openmp", when="@0.3.0:+openmp")
-    depends_on("kokkos@3.1:+cuda", when="@0.3.0:+cuda")
+    versions = {
+        ":0.2.0": "-legacy",
+        "0.3.0": "@3.1",
+        "0.4.0": "@3.2:"
+    }
+    for version in versions:
+        kk_version = versions[version]
+        for backend in kokkos_backends:
+            depends_on('kokkos{0}+{1}'.format(kk_version, backend),
+                       when='@{0}+{1}'.format(version, backend))
+    depends_on("arborx", when="@0.3.0:+arborx")
+    depends_on("heffte@2.0:", when="@0.4.0:+heffte")
+    depends_on("hypre-cmake@2.22.0:", when="@0.4.0:+hypre")
     depends_on('mpi', when='+mpi')
 
+    conflicts("+rocm", when="@:0.2.0")
+    conflicts("+sycl", when="@:0.3.0")
+
     def cmake_args(self):
         options = [
             '-DCabana_ENABLE_TESTING=ON',
-            '-DCabana_ENABLE_Serial=%s'  % (
```

Comment:
Where do these options go?

Reply:
Fixed - I now only define for the previous versions